src/index.ts

```diff
@@ -171,41 +216,32 @@ export class Iterable<T> implements LibIterable<T> {
     static empty<T>(): Iterable<T> {
         return new Iterable<T>([]);
     }
+}
 
-    private *_distinctGenerator(keySelector: (item: T, index: number) => any) {
-        const set = new Set<T>();
-        let index = 0;
-        for (const item of this._source) {
-            const key = keySelector(item, index++);
-            if (!set.has(key)) {
-                set.add(key);
-                yield item;
-            }
-        }
-    }
-
-    private *_filterGenerator(filter: (item: T, index: number) => boolean) {
-        let index = 0;
-        for (const item of this._source) {
-            if (filter(item, index++)) {
-                yield item;
-            }
-        }
-    }
-
-    private *_mapGenerator<V>(selector: (item: T, index: number) => LibIterable<V>) {
-        let index = 0;
-        for (const item of this._source) {
-            yield selector(item, index++);
+/**
+ * Returns an Iterable of the specified object's keys.
+ * @param obj {any} An object
+ * @returns {Iterable<string>} The keys of the specified object.
+ */
+export function objectKeys(obj: any): Iterable<string> {
+    return new Iterable(function* () {
+        for (const key in obj) {
+            yield key;
         }
-    }
+    });
+}
 
-    private *_mapManyGenerator<V>(selector: (item: T, index: number) => LibIterable<V>) {
-        let index = 0;
-        for (const item of this._source) {
-            yield* selector(item, index++);
+/**
+ * Returns an Iterable of the specified object's values.
+ * @param obj {any} An object
+ * @returns {Iterable<any>} The values of the specified object.
+ */
+export function objectValues(obj: any): Iterable<any> {
```

Comment:
Same as objectKeys, make this a static of Iterable. Rename to values

Reply:
done